src/sparse_ci/determinant.hpp

```diff
@@ -0,0 +1,650 @@
+    double destroy_beta_bit(int n) {
+        if (not get_beta_bit(n))
+            return 0.0;
+        set_beta_bit(n, false);
+        return slater_sign_b(n);
+    }
+
+    // THE FOLLOWING FUNCTIONS ARE NOT TESTED
+
+    /// Return the dimensions for occupied alpha orbitals
+    std::vector<std::vector<int>> get_asym_occ(int norb, std::vector<int> act_mo) const;
+    /// Return the dimensions for occupied beta orbitals
+    std::vector<std::vector<int>> get_bsym_occ(int norb, std::vector<int> act_mo) const;
+    /// Return the dimensions for virtual alpha orbitals
+    std::vector<std::vector<int>> get_asym_vir(int norb, std::vector<int> act_mo) const;
+    /// Return the dimensions for virtual beta orbitals
+    std::vector<std::vector<int>> get_bsym_vir(int norb, std::vector<int> act_mo) const;
+
+    /// Return the sign for a single second quantized operator
+    /// This function ignores if bit n is set or not
+    double slater_sign_a(int n) const {
+        if constexpr (nbits == 128) {
+            // specialization for 64 + 64 bits
+            return ui64_sign(words_[0], n);
+        } else {
+            return slater_sign(n);
+        }
+    }
+
+    /// Return the sign for a single second quantized operator
+    /// This function ignores if bit n is set or not
+    double slater_sign_b(int n) const {
+        if constexpr (nbits == 128) {
+            // specialization for 64 + 64 bits
+            return ui64_sign(words_[1], n) * ui64_bit_parity(words_[0]);
+            // the lines below have a bug
+            // static constexpr size_t parity(size_t n) { return 1 - 2 * ((n & 1) == 1); }
+            // return ui64_sign(words_[1], n) * parity(ui64_bit_count(words_[0]));
+        } else {
+            return slater_sign(n + beta_bit_offset);
+        }
+    }
+
+    /// Return the sign for a pair of alpha second quantized operators
+    /// The sign depends only on the number of bits = 1 between n and m
+    /// n and m are not assumed to have any specific order
+    double slater_sign_aa(int n, int m) const {
+        if constexpr (nbits == 128) {
+            // specialization for 64 + 64 bits
+            return ui64_sign(words_[0], n, m);
+        } else {
+            return slater_sign(n, m);
+        }
+    }
+
+    /// Return the sign for a pair of beta second quantized operators
+    /// The sign depends only on the number of bits = 1 between n and m
+    /// n and m are not assumed to have any specific order
+    double slater_sign_bb(int n, int m) const {
+        if constexpr (nbits == 128) {
+            // specialization for 64 + 64 bits
+            return ui64_sign(words_[1], n, m);
+        } else {
+            return slater_sign(n + beta_bit_offset, m + beta_bit_offset);
+        }
+    }
+
+    /// Return the sign for a quadruplet of alpha second quantized operators
+    /// a+(a) a+(b) a(j) a(i) applied to this determinant
+    /// Untested, needs documentation
```

Comment:
I remember I did a lot of work to derive all of these signs for double excitations. I tested and these should be reliable.

Reply:
Yes, these are just the functions that you wrote copied in the code. I think we already used them in the other determinant classes, so I don't see any potential problem here.